mod.ts

```diff
@@ -72,7 +92,48 @@ export function denoPlugin(options: DenoPluginOptions = {}): esbuild.Plugin {
           );
           resolved = new URL(res);
         } else {
-          resolved = new URL(args.path, referrer);
+          try {
+            resolved = new URL(args.path, referrer);
+
+            //These are thrown whenever it receives a non-URL type e.g(require("./lib/route"))
+          } catch {
+            // This is For importing files from npm-packages.. Since they dint contain extensions
+            // For Example, express's index.js file has "var route = require("./lib/route")",
+            // Since route is route.js file..
+            if (args.path.startsWith(".")) {
+              const ref = npmModulesCache.get(args.importer);
+              const specifier = ref?.name;
+              const version = ref?.versionReq;
+              const moduleDirPath =
+                `${npmCache}/registry.npmjs.org/${specifier}/${version}`;
+
+                // Checking whether it is named without extension
+              if (isEntry(args.path)) {
+
+                // This try is for consider if the path is "./route" its consider
+                // to be ./route.js and also ./route/index.js , If it not found, we can 
+                // redirect to index.js 
+                try {return {
```

Comment:
I understand this is just a placeholder and doesn't cover all the cases. Ideally we should be able to use existing package that provides Node resolution (like `resolve`). Otherwise we'd have to port a lot of very complex logic

Reply:
Yeah thats what i tried before, but i was confused with that and let me fix it and you are right.